engine/sceneManager/SceneManager.cpp

```diff
@@ -6,54 +6,71 @@
 
-	//Relativity
-	testScene->transform->SetPosition(250, 250);
-	auto& row = testScene->AddChild("TopRow");
-	row.transform->SetRotation(10);
-	row.transform->SetScale(200, 50);
-	row.AddComponent<SpriteRenderable>("Kaboom", "../assets/kaboom.png", "Kaboom");
+	_inputModule.AddMouseListener([this](const MouseEvent& mouseEvent) {
+		auto& rectangle =
+				testScene->AddChild("Rectangle_" + std::to_string(mouseEvent.x) + "_" + std::to_string(mouseEvent.y));
+		rectangle.transform->SetPosition(static_cast<float>(mouseEvent.x), static_cast<float>(mouseEvent.y));
 
-	auto& tRLeft = row.AddChild("tRLeft");
-	tRLeft.AddComponent<EllipseRenderable>("tRLeftR", glm::vec4{255, 0, 0, 255}, true);
-	tRLeft.transform->SetRotation(-35);
-	tRLeft.transform->SetScale(0.25f, 1);
-	tRLeft.transform->Translate(-0.5f + tRLeft.transform->GetLocalScale().x / 2.f, 0);
+		glm::vec4 color;
+		if (mouseEvent.button == MouseInput::BUTTON_LEFT) {
+			color = glm::vec4(255.f, 0.f, 0.f, 255.f);
+		} else if (mouseEvent.button == MouseInput::BUTTON_RIGHT) {
+			color = glm::vec4(0.f, 0.f, 255.f, 255.f);
+		} else if (mouseEvent.button == MouseInput::BUTTON_MIDDLE) {
+			color = glm::vec4(0.f, 255.f, 0.f, 255.f);
+		}
 
-	auto& tRRight = row.AddChild("trRight");
-	tRRight.AddComponent<RectangleRenderable>("trRightR", glm::vec4{0, 255, 0, 255}, true);
-	tRRight.transform->SetRotation(25);
-	tRRight.transform->SetScale(0.5f, 1);
-	tRRight.transform->Translate(0.5f - tRRight.transform->GetLocalScale().x / 2.f, 0);
+		if (mouseEvent.state == MouseButtonState::BUTTON_DOWN) {
+			rectangle.AddComponent<RectangleRenderable>("rectRenderable", color, true);
+			rectangle.transform->SetScale(20.f, 20.f);
+		} else {
+			rectangle.AddComponent<EllipseRenderable>("ellipseRenderable", color, true);
+			rectangle.transform->SetScale(20.f, 20.f);
+		}
 
-	auto& test = tRRight.AddChild("Test");
-	test.transform->SetRotation(15);
-	test.AddComponent<RectangleRenderable>("TestR", glm::vec4{0, 0, 255, 255});
+	});
 
-	//Animations
+
+
+	// Animations
 	auto& animatedObject = testScene->AddChild("AnimatedObject");
 	animatedObject.transform->SetScale(200.f, 200.f);
 	animatedObject.transform->SetPosition(100.f, 100.f);
-	animatedObject.transform->SetRotation(35);
-
 	auto& animatedSprite = animatedObject.AddComponent<AnimationRenderable>(
 			"animTag", "../assets/character_spritesheet.png",
 			"spriteTag", 32, 32
 	);
 	auto& animationController = animatedObject.AddComponent<AnimationController>("animControllerTag", animatedSprite);
 
-	animationController.AddAnimation("idle", 0, 7, 0.1f, true);
-	animationController.AddAnimation("run", 8, 15, 0.1f, true);
+	animationController.AddAnimation("idle", 0, 11, 0.15f, true);
+	animationController.AddAnimation("run", 12, 19, 0.1f, true);
+	animationController.AddAnimation("jump", 27, 35, 0.1f, false);
 
 	animatedObject.GetComponent<AnimationController>()->PlayAnimation("idle");
+
+	_inputModule.AddKeyListener([&animationController](KeyEvent event) {
```

Comment:
Graag dit niet in de SceneManager doen, gezien die überhaupt een overhaul gaat krijgen. Pls een kleine component maken in components/examples ofzo waar je dit dan doet. (die component zou je dan als sibling aan dezelfde GO hangen als die AnimationController)